File: lua/index_test.lua
-- Placeholder implementation for two_sum
function two_sum(nums, target)
    return nil
end

local function test_two_sum()
    local cases = {
        {nums = {1, 2, 3}, target = 3, expected = {0, 1}},
        {nums = {1, 3, 2}, target = 3, expected = {0, 2}},
        {nums = {30, 15, 10, 15, 30}, target = 30, expected = {1, 3}},
    }

    for _, case in ipairs(cases) do
        local result = two_sum(case.nums, case.target)
        if not result or #result ~= #case.expected then
            error(string.format("test failed: expected %s, got %s", tostring(case.expected), tostring(result)))
        end
        for i = 1, #result do
            if result[i] ~= case.expected[i] then
                error(string.format("test failed at index %d: expected %d, got %d", i, case.expected[i], result[i]))
            end
        end
    end
    print("All tests passed!")
end

test_two_sum()

File: go/index_test.go
package main

import (
	"reflect"
	"testing"
)

// Import the implementation here.
// Since I don't see a go file yet, I'll assume it will be in the same package or imported.
// For now, I'll define a placeholder for the purpose of creating the test structure.
func twoSum(nums []int, target int) []int {
	// Placeholder implementation - this will be replaced by the actual logic
	return nil
}

func TestTwoSum(t *testing.T) {
	tests := []struct {
		nums   []int
		target int
		want   []int
	}{
		{[]int{1, 2, 3}, 3, []int{0, 1}},
		{[]int{1, 3, 2}, 3, []int{0, 2}},
		{[]int{30, 15, 10, 15, 30}, 30, []int{1, 3}},
	}

	for _, tt := range tests {
		got := twoSum(tt.nums, tt.target)
		if !reflect.DeepEqual(got, tt.want) {
			t.Errorf("twoSum(%v, %d) = %v; want %v", tt.nums, tt.target, got, tt.want)
		}
	}
}
